comp/dogstatsd/server/server.go

```diff
@@ -850,6 +857,21 @@ func (s *server) parseMetricMessage(metricSamples []metrics.MetricSample, parser
 		return metricSamples, err
 	}
 
+	switch sample.metricType {
+	case gaugeType:
+		s.tlmMetricTypes.Inc("gauge")
```

Comment:
`Inc` with a value means a mutex lock and a hashmap lookup. This used to become a bottleneck in the past. Would it be possible to create `SimpleCounter` for each type and perform `Inc` on those (like [here](https://github.com/DataDog/datadog-agent/blob/81b36c3fd2c2d495340c30b1cb5042d30aca4aec/comp/dogstatsd/server/server.go#L820-L821))?

Reply:
Sure thing. I like that the SimpleCounter approach makes the cardinality explicit. 

You did nerd snipe me some. I got curious and went through the `SimpleCounter` et al implementation. I believe it's strongly consistent atomics all the way through. If we did some relaxed adds we'd be ahead on ARM. We're mutex-equivalent on both x86 and ARM right now. SimpleCounter does come out ahead on caching the hashmap lookups.